### ISSUE TITLE
Terraform sometimes creates a Heroku Cert, then gets a conflict error

### ISSUE THREAD

**Issue by JamesBelchamber** [terraform-providers/terraform-provider-heroku#37]
### Terraform Version
Terraform v0.11.1
provider.heroku v0.1.2

### Affected Resource(s)
heroku_cert

### Terraform Configuration Files
```
resource "heroku_cert" "ssl_certificate" {
  app               = "${heroku_app.default.name}"
  certificate_chain = "${file("server.crt")}"
  private_key       = "${file("server.key")}"
  depends_on        = "heroku_addon.ssl"
}
```

### Expected Behavior
Terraform successfully creates a Heroku Cert resource and records it in the state file.

### Actual Behavior
Intermittently, when creating a Heroku Cert resource, we get this error:
```
* heroku_cert.cert: Error creating SSL endpoint: Post https://api.heroku.com/apps/my-super-application/ssl-endpoints: Conflict
```

I can confirm that this ssl certificate doesn't exist prior to the Terraform run, is not duplicated in the Terraform code and does exist after the run has complete - almost like Heroku is creating the resource, then telling us it already exists(!).

I can also confirm that deleting this resource from Heroku, then allowing Terraform to recreate it, works. Importing the resource also works as a mitigation of this issue.

**Comment by Zeneixe** [comment]
👍

**Comment by bernerdschaefer** [comment]
Does the `heroku_app.default` app already exist on the first run where the cert is created?  There is a known case where creating a new app and adding a certificate very quickly (as terraform may do) can fail in this way.

You can work around this by pausing for a bit between creating an app and creating its certs:

```hcl
  provisioner "local-exec" {
    command = "echo 'Waiting for app to be ready for new certificates...'; sleep 15"
  }
```

**Comment by JamesBelchamber** [comment]
The app was new, so this fits the description. So I'm guessing this is an issue with the Heroku Platform API?

**Comment by joestump** [comment]
@bernerdschaefer we could wrap this resource in a retry. I had to do this recently with another resource. I'll take a look when I can.

**Comment by talbright** [comment]
Fixed via #142
